src/Job/DeveloperSync.php

```diff
@@ -130,21 +131,42 @@ protected function executeRequest() {
   /**
    * {@inheritdoc}
    */
-  public function execute() : bool {
+  public function execute(): bool {
     parent::execute();
 
+    // Update Apigee Edge developers and Drupal users if needed.
+    $identical_accounts = array_intersect($this->edgeAccounts, $this->drupalAccounts);
+    foreach ($identical_accounts as $search => $mail) {
+      /** @var \Apigee\Edge\Api\Management\Entity\DeveloperInterface $developer */
+      $developer = Developer::load($mail);
+      /** @var \Drupal\user\UserInterface $account */
+      $account = user_load_by_mail($mail);
+      $last_modified_delta = $developer->getLastModifiedAt()->getTimestamp() - $account->getChangedTime();
+      if ($last_modified_delta >= 0) {
+        $updateUserJob = new UserUpdate($mail);
+        $updateUserJob->setTag($this->getTag());
+        $this->scheduleJob($updateUserJob);
+      }
+      elseif ($last_modified_delta < 0) {
+        $updateDeveloperJob = new DeveloperUpdate($mail);
+        $updateDeveloperJob->setTag($this->getTag());
+        $this->scheduleJob($updateDeveloperJob);
+      }
+    }
+
+    // Create missing Drupal users.
     foreach ($this->edgeAccounts as $search => $mail) {
       if (empty($this->drupalAccounts[$search])) {
-        $createUserJob = new CreateUser($mail);
+        $createUserJob = new UserCreate($mail);
         $createUserJob->setTag($this->getTag());
         $this->scheduleJob($createUserJob);
       }
     }
 
+    // Create missing Apigee Edge developers.
     foreach ($this->drupalAccounts as $search => $mail) {
       if (empty($this->edgeAccounts[$search])) {
-        $jobs = new JobList(TRUE);
-        /** @var \Drupal\user\Entity\User $account */
+        /** @var \Drupal\user\UserInterface $account */
         if (!($account = user_load_by_mail($mail))) {
```

Comment:
If all users loaded to an array above you do not need to load one by email here probably, you can find it in the array if it is indexed by email.